src/ldp/http/metadata/ContentTypeParser.ts

```diff
@@ -7,11 +7,19 @@ import type { MetadataParser } from './MetadataParser';
  * Currently only stores the media type and ignores other parameters such as charset.
  */
 export class ContentTypeParser implements MetadataParser {
+  public static getMediaType(request: HttpRequest): string {
+    const contentType = request.headers['content-type'];
+    if (!contentType) {
+      return '';
+    }
+    return /^[^;]*/u.exec(contentType)![0].trim();
```

Comment:
The regex can trim for you

Reply:
I'm not too sure how to do that properly. Just pushed the new version that doesn't modify the `ContentTypeParser` and relies on metadata.

Reply:
If still relevant:

```typescript
     return /^[^\s;]*/u.exec(contentType)![0]; 
```